src/main/java/org/mbari/cthulhu/ui/player/MediaPlayerTimer.java

```diff
@@ -51,48 +51,77 @@
      */
-    private long previousSysMillis = 0;
+    private long previousSysMillisForMp = 0;
 
     /**
      * This is either +1 or -1 indicating time direction according to sign of
      * reported mediaPlayer time difference.
      */
     private long timeDirection;
-
+    
+    /**
+     * Was media playing in previous call to the timer?
+     */
+    private boolean wasPlaying = false;
+    
+    /**
+     * Captures system time when timer was called. Only actually used for logging, this
+     * helps visualize how accurate the frequency of the calls are wrt intended PERIOD.
+     */
+    private long previousSysMillis = 0;
+    
     /**
      * Create a timer.
      *
      * @param mediaPlayer media player that generates time-changed events
-     * @param onTick function to execute on each tick of the timer depending of time of previous call
+     * @param onTick function to execute on each tick of the timer
      */
     MediaPlayerTimer(MediaPlayer mediaPlayer, Consumer<Long> onTick) {
         timer.scheduleAtFixedRate(new TimerTask() {
             @Override
             public void run() {
+                final long newMpMillis = mediaPlayer.status().time();
+                
                 if (mediaPlayer.status().isPlaying()) {
-                    final long newMpMillis = mediaPlayer.status().time();
+                    wasPlaying = true;
+                    
+                    final long currSysMillis = System.currentTimeMillis();
                     
                     // the value to use for onTick.accept
                     long toCallTickMillis;
-    
+
                     if (newMpMillis != previousMpMillis) {
                         // mediaPlayer has reported a new value.
                         timeDirection = newMpMillis >= previousMpMillis ? +1 : -1;
                         toCallTickMillis = previousMpMillis = newMpMillis;
-                        previousSysMillis = System.currentTimeMillis();
+                        previousSysMillisForMp = currSysMillis;
                     }
                     else {
                         // mediaPlayer has reported the same value as before.
                         // Let's use last known time direction and system time difference for the onTick call:
                         toCallTickMillis = previousMpMillis +
-                            timeDirection * (System.currentTimeMillis() - previousSysMillis);
+                            timeDirection * (currSysMillis - previousSysMillisForMp);
                     }
                     
-                    log.trace("onTick: toCallTickMillis={}  timeDirection={}", toCallTickMillis, timeDirection);
+                    if (log.isTraceEnabled()) {
+                        final long sysDiffMillis = currSysMillis - previousSysMillis;
+                        previousSysMillis = currSysMillis;
+                        log.trace("onTick: toCallTickMillis={} newMpMillis={} timeDirection={} (sysDiff={})",
```

Comment:
I would comment out this entire logging block before accepting the PR. I'm concerned about the logging overhead in such a time-sensitive location.